framework/src/userobject/ActivateElementsByPath.C

```diff
@@ -0,0 +1,50 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ActivateElementsByPath.h"
+#include "libmesh/point.h"
+
+registerMooseObject("MooseApp", ActivateElementsByPath);
+
+InputParameters
+ActivateElementsByPath::validParams()
+{
+  InputParameters params = ActivateElementsUserObjectBase::validParams();
+
+  params.addParam<FunctionName>("function_x", "The x component of the heating spot travel path");
+  params.addParam<FunctionName>("function_y", "The y component of the heating spot travel path");
+  params.addParam<FunctionName>("function_z", "The z component of the heating spot travel path");
```

Comment:
Dewen, I'm sorry, I should probably have suggested to just add a default value of `0` for the function.  Then you don't need to use pointer and pointer checking. This could be done later though, as it won't change the interface.

```suggestion
  params.addParam<FunctionName>("function_x", "0", "The x component of the heating spot travel path");
  params.addParam<FunctionName>("function_y", "0", "The y component of the heating spot travel path");
  params.addParam<FunctionName>("function_z", "0", "The z component of the heating spot travel path");
```